### PR TITLE
Creating SMTP provider

### LINKED ISSUE #28494: Creating SMTP provider
### Body

Currently we have [send_email](https://github.com/apache/airflow/blob/main/airflow/utils/email.py) as util function and [EmailOperator](https://github.com/apache/airflow/blob/main/airflow/operators/email.py) as part of core.
I think email/smtp is tightly coupled to core for historical reasons but we can decouple it without making a breaking change.

**The task:**
Investigate/Implement the option of creating smtp provider

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by hussein-awala**
closes: #28494 

---
This PR creates a new provider SMTP to manage the connections, hooks and operators related to SMTP/email sending.

Once this provider is released, a second PR will be created to add it as pre-installed provider, deprecate the `airflow.operators.email.EmailOperator` and let it extends the new one with a support for Airflow config for b/c, and provide a new implementation for email backend (based on the hook).

### CODE REVIEW

**Comment by eladkal** [comment]
I think this provider will also need to be added to the pre installed ones:
https://github.com/apache/airflow/blob/275ef6cca58d9628ce1ffaeba4a416fd4a1073e6/setup.py#L690-L699

This is because currently the ability to send email is coupled with core and assuming we will deprecate the core operator in favor of the provider one we need the provider to be installed with Airflow.

**Comment by hussein-awala** [comment]
> I think this provider will also need to be added to the pre installed ones

@eladkal I cannot add before releasing the provider, where the Airflow setup uses PyPI package and not local one, so as I mentioned in the description, I will create a new PR to do that once the provider is released.

File: airflow/providers/smtp/CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    under the License.
+
+
+.. NOTE TO CONTRIBUTORS:
+   Please, only add notes to the Changelog just below the "Changelog" header when there are some breaking changes
+   and you want to add an explanation to the users on how they are supposed to deal with them.
+   The changelog is updated and maintained semi-automatically by release manager.
+
+Changelog
+---------
```
Comment (eladkal) [inline airflow/providers/smtp/CHANGELOG.rst:25]:
```suggestion
Changelog
---------

1.0.0
.....

Initial version of the provider.
```

This should be fine now

File: docs/apache-airflow-providers-smtp/connections/smtp.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    Specify the SMTP host url.
+
+Port
+    Specify the SMTP port to connect to. The default depends on the whether you use ssl or not.
+
+Extra (optional)
+    Specify the extra parameters (as json dictionary)
+
+    * ``from_email``: The email address from which you want to send the email.
+    * ``disable_ssl``: If set to true, then a non-ssl connection is being used. Default is false. Also note that changing the ssl option also influences the default port being used.
```
Comment (eladkal) [inline docs/apache-airflow-providers-smtp/connections/smtp.rst:58]: Not mandatory but would be nice to have it as boolean field in the UI

File: airflow/providers/smtp/provider.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+package-name: apache-airflow-providers-smtp
+name: Simple Mail Transfer Protocol (SMTP)
+
+description: |
+  `Simple Mail Transfer Protocol (SMTP) <https://tools.ietf.org/html/rfc5321>`__
+
+versions:
+  - 1.0.0
+
+dependencies: []
```
Comment (eladkal) [inline airflow/providers/smtp/provider.yaml]: We need minimum Airflow 2.3.0 version (as we have for all providers)

File: docs/apache-airflow-providers-smtp/connections/smtp.rst (hunk 2 above)
Comment (hussein-awala) [reply docs/apache-airflow-providers-smtp/connections/smtp.rst:58]: I already added it ([line](https://github.com/apache/airflow/pull/29968/files#diff-263bb81fd2d2743284a2b74545d6764328eb662a09f97ccc13a1126e0f99d8b5R137))

**Review by eladkal — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
LGTM

**Comment by potiuk** [comment]
Nice one @hussein-awala !

**Pull request merged**